pkg/operation/botanist/botanist.go

```diff
@@ -111,18 +125,14 @@ func (b *Botanist) RegisterAsSeed() error {
 
 // UnregisterAsSeed unregisters a Shoot cluster as a Seed in the Garden cluster.
 func (b *Botanist) UnregisterAsSeed() error {
-	seed, err := b.K8sGardenClient.GardenClientset().GardenV1beta1().Seeds().Get(b.Shoot.Info.Name, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		return nil
-	}
-	if err != nil {
-		return err
-	}
+	shootName := b.Shoot.Info.Name
 
-	if err := b.K8sGardenClient.GardenClientset().GardenV1beta1().Seeds().Delete(seed.Name, nil); err != nil && !apierrors.IsNotFound(err) {
-		return err
+	if err := b.K8sGardenClient.GardenClientset().GardenV1beta1().Seeds().Delete(shootName, nil); err != nil && !apierrors.IsNotFound(err) {
+		// Don't try to endlessly delete a Seed if it has some Shoot clusters assign to it
+		b.Logger.Errorf("Could not unregister '%s' as Seed. This error will be ignored: '%s'", shootName, err.Error())
+		return nil
```

Comment:
I don't understand why errors are ignored? If the 'not found' error is returned, sure, then we can ignore it (as it was before per condition). However, why should we ignore other errors? I don't understand the comment. The delete call itself won't fail/return an error if the Seed has Shoot's assigned, is it? The finalizer is set, and the delete call would only set the deletion time stamp?

Reply:
It is not possible to delete a Seed cluster if there are Shoot clusters assigned to it. The API server returns an error when trying to delete it.

Reply:
Where is that error coming from?

Reply:
Was the behavior changed? After rebase it's possible to delete it.

Reply:
Nope, don't think so. Not sure where your commits where based on.